packages/ai-jsx/src/core/core.ts

```diff
@@ -298,7 +315,22 @@ async function* renderStream(
       return inProgressRender;
     });
 
-    const currentValue = () => inProgressRenders.flatMap((r) => r.currentValue);
+    const currentValue = () => {
+      if (!appendOnly) {
```

Comment:
minor: I think it's generally more readable to have positive assertions, so I'd flip this to:

```tsx
if (appendOnly) { /* ... */ }

return inProgressRenders.flatMap((r) => r.currentValue);
```

Reply:
I agree

(Though, all things equal I also prefer "short blocks first" especially if I can't indent the mutually exclusive else block BECAUSE OF LINT RULES! 🤪)

Reply:
I agree "short blocks first" is also a good heuristic.

Boo hiss upon vacuous `else`. :smile: